misc/python/materialize/docker.py

```diff
@@ -16,6 +16,7 @@
 
 EXISTENCE_OF_IMAGE_NAMES_FROM_EARLIER_CHECK: dict[str, bool] = dict()
 IMAGE_TAG_OF_VERSION_PREFIX = "v"
+# XXX TODO
```

Comment:
@def-: I could use your help adapting this file. The ability to take a commit hash to a Docker image tag is pretty baked in here; unfortunately we lose that ability with this change since it's no longer possible to figure out the Docker tag for a commit hash with _just_ the commit hash, since you don't know the version in the Cargo.toml at that commit.

Reply:
I will try to help here

Reply:
I added https://github.com/MaterializeInc/materialize/pull/29097/files/a7a6ff61ec543476606fe1148002f4a27347b450..38c655d174a30e073f9ad476cc6c574ce17d751d.

Please review
* https://github.com/MaterializeInc/materialize/pull/29097/commits/38a6562a65ccce9c6046ddd1ef9448b11e896c39
* https://github.com/MaterializeInc/materialize/pull/29097/commits/38c655d174a30e073f9ad476cc6c574ce17d751d

(CC @def-)